tools/transform.cpp

```diff
@@ -266,27 +266,34 @@ static void check_refinement(Errors &errs, Transform &t,
   // so order here matters
   src_state.startParsingPre();
   expr pre = t.precondition ? t.precondition->toSMT(src_state) : true;
-  pre &= src_state.getPre() && tgt_state.getPre();
+  pre &= tgt_state.getPre() && src_state.getPre()
+          && tgt_state.getPreOnQuantifiers();
+  expr srcpre_quant = src_state.getPreOnQuantifiers();
 
   auto [poison_cnstr, value_cnstr] = type.refines(a, b);
   expr pre_dom = pre && (dom_a && dom_b);
```

Comment:
this variable doesn't seem to be used anymore

Reply:
Fixed